tools/testing/bootstrap_node_tests.ts

```diff
@@ -0,0 +1,32 @@
+import 'core-js/es7/reflect';
+import 'zone.js/dist/zone-node.js';
+import 'zone.js/dist/long-stack-trace-zone.js';
+import 'zone.js/dist/proxy.js';
+import 'zone.js/dist/sync-test.js';
+import 'zone.js/dist/async-test.js';
+import 'zone.js/dist/fake-async-test.js';
+import { TestBed } from '@angular/core/testing';
+import {
+  ServerTestingModule,
+  platformServerTesting,
+} from '@angular/platform-server/testing';
+
+// This hack is needed to get jasmine, node and zone working inside bazel.
+// 1) we load `jasmine-core` which contains the ENV: it, describe etc...
+const jasmineCore: any = require('jasmine-core');
+// 2) We create an instance of `jasmine` ENV.
+const patchedJasmine = jasmineCore.boot(jasmineCore);
+// 3) Save the `jasmine` into global so that `zone.js/dist/jasmine-patch.js` can get a hold of it to
+// patch it.
+(global as any)['jasmine'] = patchedJasmine;
+// 4) Change the `jasmine-core` to make sure that all subsequent jasmine's have the same ENV,
+// otherwise the patch will not work.
+//    This is needed since Bazel creates a new instance of jasmine and it's ENV and we want to make
+//    sure it gets the same one.
+jasmineCore.boot = function() {
+  return patchedJasmine;
+};
+
+require('zone.js/dist/jasmine-patch.js');
+
+TestBed.initTestEnvironment(ServerTestingModule, platformServerTesting());
```

Comment:
maybe in this spot you could install jasmine lifecycle hooks so that you don't need to worry about resetting the TestBed and whatnot?
I've used https://jasmine.github.io/2.1/custom_reporter.html before (but it's smelly to have a reporter messing with test state... maybe there's another way?)

Reply:
I got around the issue for now by overriding Angular's `TestBed.configureTestingModule` to always reset the testing module before reconfiguring it.